packages/plugin-ext/src/plugin/preference-registry.ts

```diff
@@ -174,22 +209,33 @@ export class PreferenceRegistryExtImpl implements PreferenceRegistryExt {
         return readonlyProxy(data);
     }
 
-    private parse(data: any): any {
-        return Object.keys(data).reduce((result: any, key: string) => {
-            const parts = key.split('.');
-            let branch = result;
-            for (let i = 0; i < parts.length; i++) {
-                if (i === parts.length - 1) {
-                    branch[parts[i]] = data[key];
-                    continue;
-                }
-                if (!branch[parts[i]]) {
-                    branch[parts[i]] = {};
+    private parse(data: PreferenceData): Configuration {
+        const defaultConfigurationModel = new ConfigurationModelImpl(this.parseConfigurationData(data[PreferenceScope.Default]), Object.keys(data[PreferenceScope.Default]));
+        const userConfiguration = new ConfigurationModelImpl(this.parseConfigurationData(data[PreferenceScope.User]), Object.keys(data[PreferenceScope.User]));
+        const workspaceConfiguration = new ConfigurationModelImpl(this.parseConfigurationData(data[PreferenceScope.Workspace]), Object.keys(data[PreferenceScope.Workspace]));
+        return new Configuration(defaultConfigurationModel, userConfiguration, workspaceConfiguration);
+    }
+
+    private parseConfigurationData(data: { [key: string]: any }): { [key: string]: any } {
+        let res;
+        try {
+            res = Object.keys(data).reduce((result: any, key: string) => {
+                const parts = key.split('.');
+                let branch = result;
+                for (let i = 0; i < parts.length; i++) {
+                    if (i === parts.length - 1) {
+                        branch[parts[i]] = data[key];
+                        continue;
+                    }
+                    if (!branch[parts[i]]) {
+                        branch[parts[i]] = {};
+                    }
+                    branch = branch[parts[i]];
                 }
-                branch = branch[parts[i]];
-            }
-            return result;
-        }, {});
+                return result;
+            }, {});
+        } catch (e) { }
```

Comment:
silent catch ?

Reply:
It fails if `data` is `undefined` or `null`.
But, in this case 
https://github.com/theia-ide/theia/blob/b8d8490740d50a97e32618d2d4d4711e49156ac2/packages/plugin-ext/src/plugin/preference-registry.ts#L213-L215
will fail as well. So, I'm going to refactor it.